docs/samples/storage/storageSpec/README.md

```diff
@@ -0,0 +1,103 @@
+# Predict on a InferenceService with StorageSpec
+## Setup
+1. Your ~/.kube/config should point to a cluster with [KServe installed](https://github.com/kserve/kserve#installation).
+2. Your cluster's Istio Ingress gateway must be [network accessible](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/).
+3. Your cluster's Istio Egresss gateway must [allow accessing S3 Storage](https://knative.dev/docs/serving/outbound-network-access/)
+
+## Create Common Storage Secret
+Create a common secret with your [storage credential](https://console.aws.amazon.com/iam/home#/users), `KServe` reads the secret key to inject
+the necessary storage volume on storage initializer or model agent to download the models from the destination storage.
+```yaml
+apiVersion: v1
+data:
+  localMinIO: ewogICJ0eXBlIjogInMzIiwKICAiYWNjZXNzX2tleV9pZCI6ICJtaW5pbyIsCiAgInNlY3JldF9hY2Nlc3Nfa2V5IjogIm1pbmlvMTIzIiwKICAiZW5kcG9pbnRfdXJsIjogImh0dHA6Ly9taW5pby1zZXJ2aWNlLmt1YmVmbG93OjkwMDAiLAogICJkZWZhdWx0X2J1Y2tldCI6ICJtbHBpcGVsaW5lIiwKICAicmVnaW9uIjogInVzLXNvdXRoIgp9
+kind: Secret
+metadata:
+  name: storage-config
+type: Opaque
+```
+
+The above `localMinIO` key is encoded with the Kubeflow minio storage credentials. The format of the above credentials is:
+```
+{
+  "type": "s3",
+  "access_key_id": "minio",
+  "secret_access_key": "minio123",
+  "endpoint_url": "http://minio-service.kubeflow:9000",
+  "default_bucket": "mlpipeline",
+  "region": "us-south"
+}
+```
+
+Apply the secret and service account
+```bash
+kubectl apply -f common_secret.yaml
+```
+
+Then, download the [sklearn model](https://storage.cloud.google.com/kfserving-samples/models/sklearn/iris/model.joblib) and store the model at the path `sklearn/model.joblib` inside the a new bucket called `example-models`.
+
+Note: if you are running kserve with istio sidecars enabled, there can be a race condition between the istio proxy being ready and the agent pulling models.
+This will result in a `tcp dial connection refused` error when the agent tries to download from s3.
+To resolve it, istio allows the blocking of other containers in a pod until the proxy container is ready.
+You can enabled this by setting `proxy.holdApplicationUntilProxyStarts: true` in `istio-sidecar-injector` configmap,
+`proxy.holdApplicationUntilProxyStarts` flag was introduced in Istio 1.7 as an experimental feature and is turned off by default.
+
+## Create the InferenceService
+Create the InferenceService with the storage spec and select the credential entry.
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: example-sklearn-isvc
+spec:
+  predictor:
+    sklearn:
+      storage:
+        key: localMinIO # Credential key for the destination storage in the common secret
```

Comment:
Do we provide a default key name?

Reply:
We don't have a default storage key for standalone deployment. Maybe I can add something for the Kubeflow deployment and borrow the KFP minio to store our models.

Reply:
I add a default key for the common secret. We can have the default key per storage in next phase when we migrate the storage initializer to new golang based image.